internal/mapset/immutable_test.go

```diff
@@ -0,0 +1,166 @@
+package mapset
+
+import (
+	"encoding/json"
+	"slices"
+	"testing"
+
+	"github.com/cedar-policy/cedar-go/internal/testutil"
+)
+
+func immutableHashSetMustNotContain[T comparable](t *testing.T, s ImmutableMapSet[T], item T) {
+	testutil.FatalIf(t, s.Contains(item), "set %v unexpectedly contained item %v", s, 1)
+}
+
+func TestImmutableHashSet(t *testing.T) {
+	t.Run("empty set contains nothing", func(t *testing.T) {
+		s := Immutable[int]()
+		testutil.Equals(t, s.Len(), 0)
+		immutableHashSetMustNotContain(t, s, 1)
+	})
+
+	t.Run("one element", func(t *testing.T) {
+		s := Immutable[int](1)
+		testutil.Equals(t, s.Contains(1), true)
+	})
+
+	t.Run("two elements", func(t *testing.T) {
+		s := Immutable[int](1, 2)
+		testutil.Equals(t, s.Contains(1), true)
+		testutil.Equals(t, s.Contains(2), true)
+		testutil.Equals(t, s.Contains(3), false)
+	})
+
+	t.Run("deduplicate elements", func(t *testing.T) {
```

Comment:
If there is a .Len() method this might be a better place to prove deduplication.